base/utf16.jl

```diff
@@ -128,7 +530,7 @@ convert(T::Type{UTF16String}, data::AbstractArray{Int16}) =
 
 function convert(T::Type{UTF16String}, bytes::AbstractArray{UInt8})
     isempty(bytes) && return UTF16String(UInt16[0])
-    isodd(length(bytes)) && throw(ArgumentError("odd number of bytes"))
+    isodd(length(bytes)) && throw(UnicodeError("odd number of bytes"))
```

Comment:
No, I don't think that's good since it would leave the code in a fairly inconsistent state. It would be better to leave the `UnicodeError` change out of this PR altogether and change all of the string types at once.